org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/services/IEditorServicesStrategy.java

```diff
@@ -34,4 +36,14 @@
      */
     FailOrSuccessResult<StyleResult, IResult> highlight(String source);
 
+    /**
+     * Attempts to provide pretty printing of the input.
+     *
+     * @param input
+     *            the term that must be pretty printed.
+     * @return {@link FailOrSuccessResult} - A {@link PrintResult} containing a human readable
+     *         text plus fold regions, or a failed result.
+     */
+    FailOrSuccessResult<PrintResult, IResult> print(ISpoofaxTermResult<?> input);
```

Comment:
The idea is that a printresult is (potentially) foldable and the `print` logic understands the foldregions. Calling this function merely asks to print a given input, thats why its just called `print`. I can't think of cases in which this method will be used without the result containing foldRegions so I renaming it is ok by me.